internal/pkg/agent/application/upgrade/artifact/download/http/downloader_test.go

```diff
@@ -253,21 +237,21 @@ func assertLogs(t *testing.T, logs []logMessage, minExpectedProgressLogs int, ex
 	// Verify that the first minExpectedProgressLogs messages are about the download progress (for the first file).
 	i := 0
 	for ; i < minExpectedProgressLogs; i++ {
-		assert.Equal(t, logs[i].record, expectedProgressMsg)
+		assert.Regexp(t, expectedProgressRegexp, logs[i].Message)
```

Comment:
If there are other logs mixed in with the progress logs, this assert will fail the test. Is this intentional?

Reply:
Good point.  As things stand, the only other log that could get mixed up is this one: https://github.com/elastic/elastic-agent/blob/359be4f18ef1104f60e9199aca268d5dcbfd77f1/internal/pkg/agent/application/upgrade/artifact/download/http/downloader.go#L102-L104

This log _should_ never get triggered but I suppose in some strange circumstances the `os.Remove` could fail. 

Any suggestions on how to make this more robust to only consider progress logs? Maybe the logs could first be filtered to only keep ones that match `expectedProgressRegexp` or `expectedCompletedRegexp` before running any assertions?

Reply:
Filtering and then asserting length is surely one option. The exact semantics depend on what we are trying to assert here: from the code you are asserting that at least the first `minExpectedProgressLogs` log entries must be the ones you expect: this will break as soon as an extra log entry is in the mix (it may even be added in the future).

Did you mean to assert that among the collected log entries you have at least `minExpectedProgressLogs`  expected logs ? In this case filter/count and then assert

Reply:
Let me put up a commit showing what I meant by filtering and see what you think.

Reply:
@pchila I implemented the filtering in a464077ad.  Let me know what you think.